File: algorithms/477. Total Hamming Distance/solution.py
class Solution(object):
    def totalHammingDistance(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        return sum(b.count('0') * b.count('1') for b in zip(*map('{:032b}'.format, nums)))

# 47 / 47 test cases passed.
# Status: Accepted
# Runtime: 298 ms

File: algorithms/477. Total Hamming Distance/solution.cc
class Solution {
public:
    int totalHammingDistance(vector<int>& nums) {
        vector<int> bits(32);
        for ( int n : nums ) {
            for ( int i = 0; i < 32; ++i ) {
                if ( n & 1 ) {
                    ++bits[i];
                }
                n >>= 1;
            }
        }
        int n = nums.size(), h = 0;
        for ( int cnt : bits ) {
            h += cnt * (n - cnt);
        }
        return h;
    }
};
